src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -296,6 +292,41 @@ internal struct NameDeclarationInfo
                 return result.Type != null;
             }
 
+            private static bool IsPatternMatching(SyntaxToken token, SemanticModel semanticModel,
+                int position, CancellationToken cancellationToken, out NameDeclarationInfo result)
+            {
+                result = default;
+                if (token.Parent.IsParentKind(SyntaxKind.IsExpression))
+                {
+                    result = IsLastTokenOfType<BinaryExpressionSyntax>(
+                        token, semanticModel,
+                        b => b.Right,
+                        _ => default(SyntaxTokenList),
+                        _ => ImmutableArray.Create(SymbolKind.Parameter),
```

Comment:
:bulb: Consider caching this `ImmutableArray<SymbolKind>` instance as a static field.